config/initializers/doorkeeper_openid_connect.rb

```diff
@@ -60,7 +67,7 @@
   # end
 
   claims do
-    claim(:email, scope: :email)     { |resource_owner| resource_owner.email }
+    claim(:email, scope: :email, response: [:user_info, :id_token]) { |resource_owner| resource_owner.email }
```

Comment:
Why is this change necessary? Shouldn't the user_info and id_token be linked to the `openid` scope instead?

Reply:
The default for `response` is `[:user_info]`. This omits the email from the generated id-token. In order to fully provision the user in nextcloud (and allow for mail-notifications), the email needs to passed in the id-token. 

I did not find a way to have one claim have different response-types depending on the scope. Also, the OpenID-scope does not pass the email.

What am I missing here?

Reply:
I could try and evaluate the scope associated with the access_token, but Doorkeeper only selects the claims which belong to the scopes associated with the access_token. I could then try an see with other scopes (and therefore claims) are present and determine further steps. 

Each claim can only be linked to one scope (at least in Doorkeeper/OIDC). Linking a central piece of the user to a not always used scope is not useful to me.

OTOH, the id_token only contains data from the claims are covered by the authorized scopes. So, if you do not want to pass your email to your OIDC-app, the scope "email" should not be part of the OAuth-Application. If the consuming app need the e-mail, it will be in the application and (hopefully) authorized by the user. With this, the consuming application receives the data directly with the JWT instead of having to ask for it via the user_info-endpoint.

So, in summary, I think the change is necessary because some OAuth-consumers only evaluate the id_token, especially nextcloud. Since no additional claim is linked to the openid-scope, there is no response to configure.